tmcprototype/CentralNode/CentralNode/CentralNode.py

```diff
@@ -426,25 +439,72 @@ def AssignResources(self, argin):
+
 
-        Note: From Jive, enter input as: {"subarrayID":1,"dish":{"receptorIDList":["0001"]}} without any space.
+            Example:
+                {
+                "dish": {
+                "receptorIDList_success": ["0001", "0002"]
+                        }
+                }
         """
+        receptorIDList = []
         try:
-            #serialize the json
+            # serialize the json
             jsonArgument = json.loads(argin)
-
-            #invoke command on subarray node
+            # Create subarray proxy
             subarrayID = int(jsonArgument['subarrayID'])
             subarrayProxy = self.subarray_FQDN_dict[subarrayID]
-            subarrayProxy.command_inout(CONST.CMD_ASSIGN_RESOURCES, jsonArgument["dish"]["receptorIDList"])
+            # Check for the duplicate receptor allocation
+            duplicate_allocation_count = 0
+            duplicate_allocation_dish_ids = []
+            for dish in range(0, len(jsonArgument["dish"]["receptorIDList"])):
+                dish_ID = "dish" + jsonArgument["dish"]["receptorIDList"][dish]
+                if self._subarray_allocation[dish_ID] != "NOT_ALLOCATED":
+                    duplicate_allocation_dish_ids.append(dish_ID)
+                    duplicate_allocation_count = duplicate_allocation_count + 1
+            if duplicate_allocation_count == 0:
+                self._resources_allocated = subarrayProxy.command_inout(CONST.CMD_ASSIGN_RESOURCES,
+                                            jsonArgument["dish"]["receptorIDList"])
+
+                # Update self._subarray_allocation variable to update subarray allocation
+                # for the related dishes.
+                # Also append the allocated dish to out argument.
+                for dish in range(0,len(self._resources_allocated)):
+                    dish_ID = "dish" + (self._resources_allocated[dish])
+                    self._subarray_allocation[dish_ID] = "SA" + str(subarrayID)
+                    receptorIDList.append(self._resources_allocated[dish])
+
+                self._read_activity_message = CONST.STR_ASSIGN_RESOURCES_SUCCESS
+            else:
+                print(CONST.STR_DISH_DUPLICATE , duplicate_allocation_dish_ids)
+                self._read_activity_message = CONST.STR_DISH_DUPLICATE + str(duplicate_allocation_dish_ids)
         except ValueError as json_exception:
             self.dev_logging(CONST.ERR_INVALID_JSON, int(tango.LogLevel.LOG_ERROR))
             self._read_activity_message = CONST.ERR_INVALID_JSON
         except KeyError as json_exception:
             self.dev_logging(CONST.ERR_JSON_KEY_NOT_FOUND, int(tango.LogLevel.LOG_ERROR))
             self._read_activity_message = CONST.ERR_JSON_KEY_NOT_FOUND
-        pass
+
+        argout = {
+            "dish": {
+                "receptorIDList_success": receptorIDList
+                        }
+                }
+        #argout['dish']['receptorIDList'] = receptorIDList
```

Comment:
remove commented code.

Reply:
Updated. The commit id is: [5d54870ec6a8497900111874538350c0dc63f179](https://github.com/ska-telescope/tmc-prototype/commit/5d54870ec6a8497900111874538350c0dc63f179)